Console/Command/CakePHPUtilShell.php

```diff
@@ -0,0 +1,330 @@
+			}
+		} else {
+			$this->_exceptions[] = 'you have to provide a SubCommand to execute!';
+		}
+
+		#TODO uncomment the line below, which echoes the help. This is used to display the getOptionsParser. Every Shell should be reformatted to the new structure first!
+		#For an example: have a look at the FeedShell
+		$this->out('-- The Help will be enabled when all Shells are restructured to the new layout -- AppShell line ' . (__LINE__ + 1));
+//        $this->out($this->getOptionParser()->help());
+		# ------------------------------------------------------------------------------------------------------------ #
+
+		# since the main() will only be called if the command does not exist, we can stop it.
+		$this->_stop();
+	}
+
+	/**
+	 * Terminates execution of the script immediately and have a look if there are exceptions whatsoever and send them to the $_mailTo
+	 *
+	 * @param int  $code     This value is used as exit code for the script (0 meaning an 'expected/healthy' termination).
+	 *
+	 * @return void
+	 */
+	protected function _stop($code = 0) {
+		# stop the timer
+		$this->_out('TIME TO COMPLETE TASK: ' . date('H\h i\m s\s', microtime(true) - $this->_startTime - 3600));
+		$this->_printWarnings();
+		// Set arg names
+		$trace = $this->_getTrace();
+		$component = ($this->command) ? $this->command : 'Unknown';
+		$args = $this->args;
+		if (!isset($this->command) || !$this->hasMethod($this->command)) {
+			# if the command is not found, or it does not exist, shift the arguments by one, because it will remain on position 0 until it has ran.
+			array_shift($args);
+		}
+		$action = implode(' ', $args);
+		$this->Mail->send($this->_mailTo, $trace[1]['class'], $component, $action, $code, $trace[0]['line'], $this->_exceptions);
+
+		// Output result to console
+		if ($code > 0 && !empty($trace['line']) && !empty($this->_exceptions)) {
+			// Unexpected shutdown, with line specification and Exceptions
+			$this->out('The ' . $trace[1]['class'] . ' has unexpectedly terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . ' of the shell script. Additionally, the following Exceptions were thrown: ' . "\n\n" . implode("\r\n", $this->_exceptions));
+		} elseif ($code > 0 && !empty($trace['line'])) {
+			// Unexpected shutdown, with line specification
+			$this->out('The ' . $trace[1]['class'] . ' has unexpectedly terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . ' of the shell script.');
+		} elseif (!empty($this->_exceptions)) {
+			// Expected shutdown with Exceptions
+			$this->out('The ' . $trace[1]['class'] . ' has terminated the \'' . $component . '/' . $action . '\' action with exit code ' . $code . '. The error occurred on line ' . $trace[0]['line'] . '. Additionally, the following Exceptions were thrown: ' . "\n\n" . implode("\r\n", $this->_exceptions));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 297 characters long line (which is greater than 120 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS103)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Combine these strings instead of concatenating them. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://dev.expandonline.nl:9000/sonarqube/coding_rules#rule_key=php%3AS2005)